sdk/tables/Azure.Data.Tables/src/TableConnectionString.cs

```diff
@@ -328,8 +328,15 @@ internal static string GetAccountNameFromUri(Uri uri)
             {
                 // This is most likely Azurite, which looks like this: https://127.0.0.1:10002/contoso/
                 var segments = uri.Segments;
-
-                accountName = segments[1].EndsWith("/", StringComparison.OrdinalIgnoreCase) ? segments[1].Substring(0, segments[1].Length - 1) : segments[1];
+                if (segments.Length == 1)
```

Comment:
Are we replacing any IPv4 address with a localhost?

Reply:
No, we're just making an assumption here that if no account name is specified in the actual TableEndpoint connection string part, and it is an ipv4 host address, they are most likely using a Cosmos emulator connection string. The Cosmos emulator connection string is a well known string, and has these properties. Any connection string specifying the account name in the TableEndpoint will always be respected.

Reply:
Should we attempt a DNS lookup in this scenario and validate that the segment resolves to 127.0.0.1?

Reply:
I don't think it makes sense to validate since any IP address would tend to be an emulator. Is there a scenario you had in mind?

Reply:
Nothing concrete; I was just thinking that if we're assuming localhost, there was a way (albeit a slow one that doesn't really fall into the scope of a parser) that could help prove it.